code/computational_geometry/src/cohen_sutherland_lineclip/Cohen-SutherlandLineClippingAlgorithm.cpp

```diff
@@ -0,0 +1,153 @@
+
+        #include <iostream>
+        // Part of Cosmos by OpenGenus Foundation
+        class CohenSutherLandAlgo
+        {
+        private:
+            double x1,y1,x2,y2;
+            double x_max,y_max,x_min,y_min;
+            const int INSIDE = 0; // 0000
+            const int LEFT   = 1; // 0001
+            const int RIGHT  = 2; // 0010
+            const int BOTTOM = 4; // 0100
+            const int TOP    = 8; // 1000
+        public:
+            CohenSutherLandAlgo()
+            {
+                x1 = 0.0;
+                x2 = 0.0;
+                y1 = 0.0;
+                y2 = 0.0;
+            }
+            void getCoordinates();
+            void getClippingRectangle();
+            int generateCode(double x, double y);
+            void cohenSutherland();
+        };
+        void CohenSutherLandAlgo::getCoordinates()
+        {
+            std::cout << "\nEnter Co-ordinates of P1(X1,Y1) of Line Segment : ";
```

Comment:
Is it possible that you could instead read these values from parameters and instead move IO bound stuff to `main`? This way, this class is easily reusable.

Reply:
Yeah it is possible..I will change the implementation